src/libcore/pin.rs

```diff
@@ -584,6 +584,25 @@ impl<P: DerefMut> Pin<P> {
     /// the pointee cannot move after `Pin<Pointer<T>>` got created.
     /// "Malicious" implementations of `Pointer::DerefMut` are likewise
     /// ruled out by the contract of `Pin::new_unchecked`.
+    ///
+    /// This method is useful when calling a function that consumes the pinned type multiple times.
+    ///
+    /// ```
+    /// use std::pin::Pin;
+    ///
+    /// # struct Type {}
+    /// impl Type {
+    ///     fn method(self: Pin<&mut Self>) {
+    ///         // do something
+    ///     }
+    ///
+    ///     fn call_method_twice(mut self: Pin<&mut Self>) {
+    ///         // `method` consumes `self`, so reborrow the `Pin<&mut Self>` via `as_mut`.
+    ///         x.as_mut().method();
```

Comment:
`x` should probably be `self` then? (For a PR like this, please run `./x.py test --stage 0 src/libcore` before asking for review.)

Reply:
Oops, sorry. Fixed in latest commit (`./x.py test --stage 0 src/libcore` failed with the following error, so I tested it with `--stage 1`)

```text
error[E0432]: unresolved import `super`

error[E0425]: cannot find value `const_unchecked_layout` in module `__test_reexports::alloc`
  --> src/libcore/../libcore/tests/alloc.rs:4:1
   |
3  |   #[test]
   |   ------- in this macro invocation
4  | / fn const_unchecked_layout() {
5  | |     const SIZE: usize = 0x2000;
6  | |     const ALIGN: usize = 0x1000;
7  | |     const LAYOUT: Layout = unsafe { Layout::from_size_align_unchecked(SIZE, ALIGN) };
8  | |     assert_eq!(LAYOUT.size(), SIZE);
9  | |     assert_eq!(LAYOUT.align(), ALIGN);
10 | | }
   | |_^ not found in `__test_reexports::alloc`
   | 
  ::: <::core::macros::builtin::test macros>:1:1
   |
1  |   ($ item : item) => { }
   |   ---------------------- in this expansion of `#[test]`
help: possible candidate is found in another module, you can import it into scope
   |
1  | use crate::alloc::const_unchecked_layout;
   |
```